qir/qat/GroupingPass/GroupingPass.cpp

```diff
@@ -309,7 +317,15 @@ namespace quantum
             auto ptr = *it;
             if (!ptr->use_empty())
             {
-                llvm::errs() << ";; Error: Could not delete " << *ptr << "\n";
+                if (logger_)
+                {
+                    logger_->setLocationFromValue(ptr);
+                    logger_->error("Could not delete node.");
+                }
+                else
+                {
+                    throw std::runtime_error("No logger present - Error: Unable to delete instruction.\n");
+                }
             }
             else
             {
```

Comment:
Feels like 2 identical loops in green lines 249 - 268 and 315 - 334. Consider de-duplicating.

Reply:
Yes, this is a good point. This has been fixed.